Core/src/main/java/com/plotsquared/core/plot/flag/PlotFlag.java

```diff
@@ -158,6 +158,15 @@ public Caption getFlagCategory() {
         return this.flagCategory;
     }
 
+    /**
+     * Get if the flag's permission should check for values. E.g. plot.flag.set.music.VALUE
+     *
+     * @return if valued permission
+     */
+    public boolean isValuedPermission() {
```

Comment:
Maybe this should be a more extensible system that e.g. also allows to check for ranges for Long/Double flags?

Reply:
See #3201

Reply:
Our own flags aren't the only ones that exist. 
What I meant is, that it would probably be worth to let the flag check if a value is accepted based on a user context, not necessarily dependent on permissions. While this would make the flags less value-based, it would be far more flexible and future-proof.

Reply:
The check is there so P2 knows if the permission `plots.flag.set.time.20000` should be looked up (it shouldn't). Similarly `plots.flag.set.greeting.Welcome to my plot!`. Whereas something like `plots.flag.set.music.cat` should be searched, removing the need for the plots.flag.set.greeting.* permission where it really shouldn't be required.